addons/hr_holidays/tests/test_leave_requests.py

```diff
@@ -807,19 +839,18 @@ def test_holiday_type_requires_no_allocation(self):
                 'date_from': '2020-01-01',
                 'date_to': '2020-12-31',
             })
-            allocation.action_validate()
             leave1 = self.env['hr.leave'].with_user(self.user_employee_id).create({
                 'name': 'Holiday Request',
                 'employee_id': self.employee_emp_id,
                 'holiday_status_id': self.holidays_type_2.id,
-                'date_from': '2020-09-06',
-                'date_to': '2020-09-08',
+                'date_from': '2020-09-07',
+                'date_to': '2020-09-10',
                 'number_of_days': 3,
             })
 
             self._check_holidays_count(
-                self.holidays_type_2.get_employees_days([self.employee_emp_id])[self.employee_emp_id][self.holidays_type_2.id],
-                ml=5, lt=0, rl=5, vrl=2, vlt=3, closest_allocation=allocation,
+                self.holidays_type_2.get_allocation_data(self.employee_emp)[self.employee_emp][0][1],
```

Comment:
Why is get_allocation_data used here and get_leaves_consumed used in the subsequent lines?